tooling/v8-snapshot/src/setup/config.ts

```diff
@@ -0,0 +1,120 @@
+import path from 'path'
+
+type SnapshotConfig = {
+  appEntryFile: string
+  cypressAppSnapshotDir: string
+  nodeModulesOnly: boolean
+  pathsMapper: (file: string) => string
+  projectBaseDir: string
+  snapshotCacheDir: string
+  snapshotEntryFile: string
+  snapshotMetaFile: string
+  snapshotMetaPrevFile: string
+  metaFile: string
+  usePreviousSnapshotMetadata: boolean
+}
+
+const platformString = process.platform
+
+const snapshotCacheBaseDir = path.resolve(__dirname, '..', '..', 'cache')
+
+const projectBaseDir = path.join(__dirname, '..', '..', '..', '..')
+const appEntryFile = require.resolve('@packages/server/server-entry.js')
+
+const cypressAppSnapshotDir = (() => {
+  const electronPackageDir = path.join(projectBaseDir, 'packages', 'electron')
+
+  let cypressApp
+  let electronPath
+
+  if (platformString === 'darwin') {
+    cypressApp = path.join('dist', 'Cypress', 'Cypress.app')
+    electronPath = path.join('Contents', 'Frameworks', 'Electron Framework.framework', 'Versions', 'A', 'Resources')
+  } else {
+    cypressApp = path.join('dist', 'Cypress')
+    electronPath = ''
+  }
+
+  const cypressAppDir = path.join(electronPackageDir, cypressApp)
+
+  return path.join(
+    cypressAppDir,
+    electronPath,
+  )
+})()
+
+const pathsMapper = (s: string) => s.replace(/^packages\//, './packages/')
+
+const usePreviousSnapshotMetadata = process.env.V8_SNAPSHOT_FROM_SCRATCH == null
+
+/**
+ * @typedef {Object} SnapshotConfig          - Configuration for creating snapshots.
+ *
+ * @property {string} appEntryFile           - the app entry file used to generate the, see snapshotEntryFile
```

Comment:
Is this what you meant?

```suggestion
 * @property {string} appEntryFile           - the app entry file used to generate the app entry, see snapshotEntryFile
```

Reply:
[cypress-io/cypress@`ec34a53` (#23732)](https://github.com/cypress-io/cypress/pull/23732/commits/ec34a537a4879edcfbc0409b0f81f16a94e97ee9)